v3/generator/client/client.tmpl

```diff
@@ -19,10 +19,14 @@ type Client struct {
 	timeout         time.Duration
 	pollingInterval time.Duration
 	trace           bool
-	middlewares     []RequestMiddlewareFn
+
+	// A list of callbacks for modifying requests which are generated before sending over
+	// the network.
+	requestEditors []RequestEditorFn
 }
 
-type RequestMiddlewareFn func(ctx context.Context, req *http.Request) error
+// RequestEditorFn  is the function signature for the RequestEditor callback function
+type RequestEditorFn func(ctx context.Context, req *http.Request) error
```

Comment:
I'm not 100% convinced with the name, perhaps we can keep the previous RequestInterceptor name even though the real behavior changed a bit (impossible to act on the response) ?

Reply:
https://github.com/exoscale/egoscale/pull/611/commits/f4c120bec71db146ecb71061a30f35cfba191e03